FlightReservation/components/modalComponent.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import {
+    StyleSheet,
+    View,
+    Text,
+    Image,
+    Modal,
+  } from 'react-native';
+import { AnimatedCircularProgress } from 'react-native-circular-progress';
+
+function ModalCustom(props) {
+    const { modalVisible, text, isIconCheck } = props;
+    return (
+      <View style={styles.centeredView}>
+        <Modal
+          animationType="slide"
+          transparent={true}
+          visible={modalVisible}
+          presentationStyle="overFullScreen"
+        >
+          <View style={[styles.centeredView, {backgroundColor:'rgba(0, 0, 0, 0.4)' }]}>
+            <View style={[styles.modalView, {backgroundColor: "black"}]}>
+              {isIconCheck
+                ? <AnimatedCircularProgress
+                  size={80}
+                  width={8}
+                  fill={100}
+                  tintColor="#5362D8"
+                  backgroundColor="#3d5875"
+                  duration={2500}
+                />
+                : <Image
+                  style={styles.iconCheck}
+                  source={require('../img/check.png')}
+                />
+              }
```

Comment:
instead of having this ternary condition, it would be better to call a function with the return of this JSX